File: examples/nodejs/faulty.js
#!/usr/bin/env node
// Faulty Node.js code with common issues

const fs = require('fs');
const http = require('http');
var path = require('path');

var data = fs.readFileSync('/tmp/data.txt', 'utf8');
console.log("Data loaded: " + data);

var config = {
    port: process.env.PORT
};

if (config.port == undefined) {
    config.port = 3000;
}

function processFile(filename) {
    var content = fs.readFileSync(filename);
    console.log("Processing: " + filename);
    
    if (content == null) {
        return;
    }
    
    var result = eval(content.toString());
    return result;
}

http.createServer(function() {
    console.log("Request received");
    var response = fs.readFileSync('./response.html');
    return response;
}).listen(config.port);

module.exports = {
    processFile: processFile
};

File: examples/python/faulty.py
#!/usr/bin/env python3
import os
import sys
import json

def process_data(items=[]):
    for item in items:
        if item == None:
            print "Item is None"
            continue
        try:
            result = item * 2
        except:
            print "Error processing item"
    return items

def calculate_average(numbers):
    total = sum(numbers)
    return total / len(numbers)

if __name__ == "__main__":
    data = [1, 2, None, 4]
    process_data(data)
    
    if len(sys.argv) == None:
        print "No arguments"
